src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -115,26 +126,42 @@ bool TestDACRevocationDelegateImpl::CrossValidateCert(const Json::Value & revoke
 bool TestDACRevocationDelegateImpl::IsEntryInRevocationSet(const std::string & akidHexStr, const std::string & issuerNameBase64Str,
                                                            const std::string & serialNumberHexStr)
 {
-    std::ifstream file(mDeviceAttestationRevocationSetPath.c_str());
-    if (!file.is_open())
-    {
-        ChipLogError(NotSpecified, "Failed to open file: %s", mDeviceAttestationRevocationSetPath.c_str());
-        return false;
-    }
-
-    // Parse the JSON data incrementally
     Json::CharReaderBuilder readerBuilder;
     Json::Value jsonData;
     std::string errs;
 
-    bool parsingSuccessful = Json::parseFromStream(readerBuilder, file, &jsonData, &errs);
+    // Try direct data first, then fall back to file
+    std::istringstream jsonStream(!mRevocationData.empty() ? mRevocationData : "[]");
```

Comment:
This is only used in the `!mRevocationData.empty()` block below.  Why is it declared up here?

Reply:
moved it to the scope that it using it.